packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -64,82 +112,236 @@ const AccountRecoveryResetPassword = ({
+
+      {/* Hidden email field is to allow Fx password manager
+        to correctly save the updated password. Without it,
+        the password manager tries to save the old password
+        as the username. */}
+      <input type="email" value={state.email} className="hidden" readOnly />
+      <section className="text-start mt-4">
+        <FormPasswordWithBalloons
+          {...{
+            formState,
+            errors,
+            trigger,
+            register,
+            getValues,
+            passwordMatchErrorText,
+            setPasswordMatchErrorText,
+          }}
+          passwordFormType="reset"
+          onSubmit={handleSubmit(
+            (data: FormData) => {
+              onSubmit(data);
+            },
+            (err) => {
+              console.error(err);
+            }
+          )}
+          email={state.email}
+          loading={false}
+          onFocusMetricsEvent={`${viewName}.engage`}
+        />
+      </section>
+
+      <LinkRememberPassword {...state} />
+    </AppLayout>
   );
+
+  /**
+   * Determines starting state for component
+   */
+  function getInitialState() {
+    let email = '';
+    let linkStatus: LinkStatus = LinkStatus.valid;
+    let forceAuth = false;
+    let supportsRecovery = true;
+    let contextError: ContextValidationErrors | null = null;
+
+    try {
+      email = verificationInfo.email || '';
+
+      forceAuth = !!verificationInfo.forceAuth;
+
+      if (!verificationInfo.isValid()) {
+        supportsRecovery = false;
+        linkStatus = LinkStatus.damaged;
+      } else if (!accountRecoveryKeyInfo.isValid()) {
+        supportsRecovery = false;
+      } else if (verificationInfo.lostRecoveryKey === true) {
+        supportsRecovery = false;
+      }
+    } catch (err) {
+      if (err instanceof ContextValidationErrors) {
+        contextError = err;
+        linkStatus = LinkStatus.damaged;
+      }
+    }
+
+    return {
+      email,
+      linkStatus,
+      forceAuth,
+      supportsRecovery,
+      contextError,
+    };
+  }
+
+  async function onSubmit(data: FormData) {
+    const password = data.newPassword;
+
+    try {
+      await account.resetPasswordWithRecoveryKey({
+        password,
+        ...verificationInfo,
+        ...accountRecoveryKeyInfo,
+      });
+
+      // FOLLOW-UP: Functionality not yet available.
+      await account.setLastLogin(Date.now());
+
+      // QUESTION: It seems like the account class will now refresh itself, so I don't think
+      //           these are necessary anymore?
+      // storageContext.set('currentAccountUid', account.uid);
+      // await account.refresh('account');
+
+      // FOLLOW-UP: Functionality not yet available.
+      notifier.onAccountSignIn(account);
+
+      relier.resetPasswordConfirm = true;
+      logViewEvent(viewName, 'verification.success');
+
+      // FOLLOW-UP: Functionality not yet available.
+      await broker.invokeBrokerMethod('afterCompleteResetPassword', account);
```

Comment:
@dschom Do you mind throwing anything needed to match parity functionality-wise into the next sprint (and/or in tickets blocking reset PW prod push)?

Reply:
@LZoog I'll go ahead and file them. BTW, If you disagree with any of the follow-up assessments let me know. Perhaps this is something that should be pulled into this ticket, but as of now, all the follow-up comments identify gaps in what is needed by this page that wasn't already supported.